mlrun/utils/helpers.py

```diff
@@ -1886,3 +1886,15 @@ def run_with_retry(
             if attempt == retry_count:
                 raise
     raise last_exception
+
+
+def join_urls(base_url: str, path: str) -> str:
+    """
+    Joins a base URL with a path, ensuring proper handling of slashes.
+
+    :param base_url: The base URL (e.g., "http://example.com").
+    :param path: The path to append to the base URL (e.g., "/path/to/resource").
+
+    :return: A unified URL with exactly one slash between base_url and path.
+    """
+    return f"{base_url.rstrip('/')}/{path.lstrip('/')}" if path else base_url
```

Comment:
why not use `urllib.parse.urljoin` instead?

Reply:
@TomerShor because it doesn't work correctly if there is no http(s):// in the base_url:
<img width="476" alt="image" src="https://github.com/user-attachments/assets/4821ed5a-016b-4ee5-ac7d-f8b72c8e67a9">

Reply:
@rokatyy You need to validate that the `base_url` is not empty

Reply:
@moranbental what do you mean?

Reply:
@rokatyy 
When `base_url = None`: The code will throw an error.
When `base_url = ""`: The code will return `"/path"` (Is that the expected behavior?)

Reply:
@moranbental type annotation require both arguments to be string, so not sure if it worth checking, but okey, will add 

reg the case when base_url is empty string, I think it's expected.